nifi-docs/src/main/asciidoc/administration-guide.adoc

```diff
@@ -2477,6 +2477,13 @@ implementation.
 |`nifi.flowfile.repository.always.sync`|If set to `true`, any change to the repository will be synchronized to the disk, meaning that NiFi will ask the operating system not to cache the information. This is very expensive and can significantly reduce NiFi performance. However, if it is `false`, there could be the potential for data loss if either there is a sudden power loss or the operating system crashes. The default value is `false`.
 |====
 
+==== Encryption
+
+The FlowFile repository can be configured to encrypt all files as they are written to disk.  To enable this encryption,
+set the `nifi.flowfile.repository.always.key.1` property to a 16 or 32 bit value like this:
```

Comment:
In the description, it indicates setting "nifi.flowfile.repository.always.key.1" whereas in the example, you use "...cipher.key.1" instead of "...always.key.1" - I'm not sure that I understand the difference. Also, is it expecting multiple keys? If so, it probably makes sense in the example to show multiple keys and to explain how the different keys may be used. If it expects only a single key be provided, why use a suffix such as `.1`?

Reply:
Great point!

Still working on the key management + documentation bits.  Will update when I've got more.

Reply:
I've added explicit key lookup to the nifi props class, and support for retrieving key material where needed.